recipes/default.rb

```diff
@@ -17,48 +17,45 @@
 # limitations under the License.
 #
 
-lang = node[:locale][:lang]
-lc_all = node[:locale][:lc_all] || lang
+lang = node['locale']['lang']
+lc_all = node['locale']['lc_all'] || lang
 
-case node['platform']
-when 'debian', 'ubuntu'
-
-  package "locales" do
+if File.exist?('/usr/sbin/update-locale')
+  # FIXME: do we really need to install this??!?!
+  package 'locales' do
     action :install
   end
 
-  execute "Generate locale" do
+  execute 'Generate locale' do
     command "locale-gen #{lang}"
-    not_if { Locale.up_to_date?("/etc/default/locale", lang, lc_all) }
+    not_if { Locale.up_to_date?('/etc/default/locale', lang, lc_all) }
   end
 
-  execute "Update locale" do
+  execute 'Update locale' do
     command "update-locale LANG=#{lang} LC_ALL=#{lc_all}"
-    not_if { Locale.up_to_date?("/etc/default/locale", lang, lc_all) }
+    not_if { Locale.up_to_date?('/etc/default/locale', lang, lc_all) }
   end
-
-when 'redhat', 'centos', 'scientific', 'amazon'
-  locale_file_path = "/etc/sysconfig/i18n"
+elsif File.exist?('/usr/bin/localectl')
```

Comment:
I'd prefer to move to putting chef-sugar's `systemd?` helper into Chef::DSL::Universal.